kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/RequestFactory.java

```diff
@@ -193,39 +197,52 @@ private static void populateOffsetFetchRequest(ApiMessage apiMessage) {
         topics.setPartitionIndexes(List.of(0, 1));
         group.setTopics(List.of(topics));
         offsetFetchRequestData.setGroups(List.of(group));
+
+        if (apiVersion <= 7) {
+            offsetFetchRequestData.setGroups(List.of());
+        }
+        else {
+            offsetFetchRequestData.setGroupId("");
+            offsetFetchRequestData.setTopics(List.of());
```

Comment:
I think what you have is correct, OffsetFetchRequestTopic isn't used 8+, but this the code isn't terribly obvious.  I think the creation of the OffsetFetchRequestTopic belongs in the `(apiVersion <= 7)` block.